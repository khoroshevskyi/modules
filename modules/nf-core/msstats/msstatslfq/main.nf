process MSSTATS_MSSTATSLFQ {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bioconductor-msstats:4.14.0--r44he5774e6_0' :
        'biocontainers/bioconductor-msstats:4.14.0--r44he5774e6_0' }"

    input:
    tuple val(meta), path(msstats_csv_input)

    output:
    // The generation of the PDFs from MSstats are very unstable, especially with auto-contrasts.
    // And users can easily fix anything based on the csv and the included script -> make optional
    tuple val(meta), path("*.pdf"), emit: pdf, optional: true
    tuple val(meta), path("*.csv"), emit: msstats_csv
    tuple val(meta), path("*.log"), emit: log
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    msstats_plfq.R \\
        ${msstats_csv_input} \\
        ${args} \\
        2>&1 | tee msstats.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-msstats: \$(Rscript -e "library(MSstats); cat(as.character(packageVersion('MSstats')))")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''

    """
    touch test_sample_msstats_in_msstats_results.csv
    touch msstats.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-msstats: \$(Rscript -e "library(MSstats); cat(as.character(packageVersion('MSstats')))")
    END_VERSIONS
    """
}
