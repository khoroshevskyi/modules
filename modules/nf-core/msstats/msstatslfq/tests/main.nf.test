nextflow_pipeline {
    name "Test MSSTATS_MSSTATSLFQ"
    script "modules/nf-core/msstats/msstatslfq/main.nf"
    test "proteomics - MSstats label-free quantification" {
        when {
            params {
                outdir = "$baseDir/results"
                contrasts = "A-B"
                ref_condition = "A"
                msstats_remove_one_feat_prot = true
                msstatslfq_removeFewMeasurements = true
                msstatslfq_feature_subset_protein = "all"
                msstatslfq_quant_summary_method = "TMP"
                msstats_threshold = 0.01
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
