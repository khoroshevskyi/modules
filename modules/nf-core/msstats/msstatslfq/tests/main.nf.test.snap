[
    {
        "0": [
            "versions.yml:md5,2164617e154ea0ee042062991af6e2d7"
        ],
        "1": [
            [
                "test_sample_msstats_in.csv",
                "test_sample_msstats_in_msstats_results.csv:md5,2164617e154ea0ee042062991af6e2d7"
            ]
        ],
        "2": [
            [
                "test_sample_msstats_in.csv",
                "msstats.log:md5,2164617e154ea0ee042062991af6e2d7"
            ]
        ],
        "versions": [
            "versions.yml:md5,2164617e154ea0ee042062991af6e2d7"
        ]
    }
]
